src/device-manager/python/openweave/WeaveTLV.py

```diff
@@ -0,0 +1,619 @@
+            decoding["value"] = False
+        elif decoding["type"] == "Unsigned Integer 1-byte value":
+            decoding["value"], = struct.unpack("<B", tlv[self._bytesRead: self._bytesRead + 1])
+            self._bytesRead += 1
+        elif decoding["type"] == "Signed Integer 1-byte value":
+            decoding["value"], = struct.unpack("<b", tlv[self._bytesRead: self._bytesRead + 1])
+            self._bytesRead += 1
+        elif decoding["type"] == "Unsigned Integer 2-byte value":
+            decoding["value"], = struct.unpack("<H", tlv[self._bytesRead: self._bytesRead + 2])
+            self._bytesRead += 2
+        elif decoding["type"] == "Signed Integer 2-byte value":
+            decoding["value"], = struct.unpack("<h", tlv[self._bytesRead: self._bytesRead + 2])
+            self._bytesRead += 2
+        elif decoding["type"] == "Unsigned Integer 4-byte value":
+            decoding["value"], = struct.unpack("<L", tlv[self._bytesRead: self._bytesRead + 4])
+            self._bytesRead += 4
+        elif decoding["type"] == "Signed Integer 4-byte value":
+            decoding["value"], = struct.unpack("<l", tlv[self._bytesRead: self._bytesRead + 4])
+            self._bytesRead += 4
+        elif decoding["type"] == "Unsigned Integer 8-byte value":
+            decoding["value"], = struct.unpack("<Q", tlv[self._bytesRead: self._bytesRead + 8])
+            self._bytesRead += 8
+        elif decoding["type"] == "Signed Integer 8-byte value":
+            decoding["value"], = struct.unpack("<q", tlv[self._bytesRead: self._bytesRead + 8])
+            self._bytesRead += 8
+        elif decoding["type"] == "Floating Point 4-byte value":
+            decoding["value"], = struct.unpack("<f", tlv[self._bytesRead: self._bytesRead + 4])
+            self._bytesRead += 4
+        elif decoding["type"] == "Floating Point 8-byte value":
+            decoding["value"], = struct.unpack("<d", tlv[self._bytesRead: self._bytesRead + 8])
+            self._bytesRead += 8
+        elif "UTF-8 String" in decoding["type"]:
+            val, = struct.unpack("<%ds" % decoding["strDataLen"],
+                                               tlv[self._bytesRead: self._bytesRead + decoding["strDataLen"]])
+            try:
+                decoding["value"] = str(val, 'utf-8')
+            except:
+                decoding["value"] = val
+            self._bytesRead += decoding["strDataLen"]
+        elif "Byte String" in decoding["type"]:
+            val, = struct.unpack("<%ds" % decoding["strDataLen"],
+                                 tlv[self._bytesRead: self._bytesRead + decoding["strDataLen"]])
+
+            decoding["value"] = val
+            self._bytesRead += decoding["strDataLen"]
+        else:
+            raise ValueError('Attempt to decode unsupported TLV type')
+
+    def _get(self, tlv, decodings, out):
+        endOfEncoding = False
+
+        while len(tlv[self._bytesRead:]) > 0 and endOfEncoding == False:
```

Comment:
For python3, this fails. Slicing only works on strings. This generates 'TypeError: 'int' object is not subscriptable'
I think casting tlv to str will work. len(str(tlv)[x:])

Reply:
if it is integer, it should be the error code returned from data management client, i have added check, we can sync offline on it, thanks